### PR TITLE
feat: snowflake query generator

### LINKED ISSUE #24781: [Feature]: one click binding support for snowflake datasource
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Add Snowflake db to the list of the supported data sources on one click binding

### Why should this be worked on?

It extends the feature to more users.

### PR DESCRIPTION
**Pull request by berzerkeer**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.
>
#### PR fixes following issue(s)
Fixes #24781 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by berzerkeer** [comment]
The base branch for this PR is `feat/one-click-mssql-mysql` because this was branched off of that branch. A [PR](https://github.com/appsmithorg/appsmith/pull/24516) is out for that branch once its merged we can merge this snowflake PR.

File: app/client/src/WidgetQueryGenerators/Snowflake/index.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
+import { BaseQueryGenerator } from "../BaseQueryGenerator";
+import { format } from "sql-formatter";
+import { QUERY_TYPE } from "../types";
+import type {
+  WidgetQueryGenerationConfig,
+  WidgetQueryGenerationFormConfig,
+  ActionConfigurationSQL,
+} from "../types";
+import { removeSpecialChars } from "utils/helpers";
+export default abstract class Snowflake extends BaseQueryGenerator {
```
Comment (sbalaji1192) [inline app/client/src/WidgetQueryGenerators/Snowflake/index.ts:10]: leave a line of space here

File: app/client/src/WidgetQueryGenerators/Snowflake/index.ts [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+      return;
+    }
+
+    const { value, where } = update;
+
+    return {
+      type: QUERY_TYPE.UPDATE,
+      name: `Update_${removeSpecialChars(formConfig.tableName)}`,
+      payload: {
+        body: `UPDATE ${formConfig.tableName} SET ${formConfig.columns
```
Comment (sbalaji1192) [inline app/client/src/WidgetQueryGenerators/Snowflake/index.ts:114]: we need to omit primary column

**Review by sbalaji1192 — COMMENTED** [review]
@berzerkeer we need to write cypress test for this.

**Comment by berzerkeer** [comment]
@sbalaji1192 no existing cypress for snowflake at all. I think its because there is no free version of snowflake . Tagging @Aishwarya-U-R for more clarity. Thanks!

**Comment by Aishwarya-U-R** [comment]
> @sbalaji1192 no existing cypress for snowflake at all. I think its because there is no free version of snowflake . Tagging @Aishwarya-U-R for more clarity. Thanks!

Yes, no snowflake since we dont have a free account/test account for the same.

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**